### PR TITLE
[eas-cli] Fix update --channel and --branch flag validation

### PR DESCRIPTION
**Pull request by byCedric**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

A partner reported an issue when using the following command combination:

```bash
$ eas update --branch "..." --message "..." --non-interactive
> Error: --auto or both --channel or --branch and --message are required
> when updating in non-interactive mode
```

# How

It seems that PR #1570 introduced an incorrect validation combination, where it requires users to provide all flags (`--branch`, `--channel`, _and_ `--message`) when not using `--auto`.

This changes:
- The validation to allow combinations like:
  - `--branch` + `--message`
  - `--channel` + `--message`
  - `--branch` + `--channel` + `--message` (this is caught after flag validation as incorrect combination)
- Change the error message in attempt to improve communicating the allowed combinations
  > ```bash
  > Error: --branch and --message, or --channel and --message are required when updating in non-interactive mode unless --auto is specified
  > ```

# Test Plan

Added basic test for this case, including some `--republish` deprecation tests.

### CODE REVIEW

**Comment by byCedric** [comment]
/changelog-entry bug-fix Fix update `--channel` and `--branch` flag validation

File: packages/eas-cli/src/commands/update/index.ts [hunk 1]
```diff
@@ -509,9 +509,9 @@ export default class UpdatePublish extends EasCommand {
     const nonInteractive = flags['non-interactive'] ?? false;
 
     const { auto, branch: branchName, channel: channelName, message: updateMessage } = flags;
-    if (nonInteractive && !auto && !(branchName && channelName && updateMessage)) {
+    if (nonInteractive && !auto && !(updateMessage && (branchName || channelName))) {
       Errors.error(
-        '--auto or both --channel or --branch and --message are required when updating in non-interactive mode',
+        '--auto or --message and either --branch or --channel are required when updating in non-interactive mode',
```
Comment (wkozyra95) [inline packages/eas-cli/src/commands/update/index.ts:514]: This is an improvement, but I think this error is still not clear. I sounds to me that (auto or message) and (channel  or branch) is required.

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/commands/update/index.ts (hunk 1 above)
Comment (byCedric) [reply packages/eas-cli/src/commands/update/index.ts:514]: Open for other suggestions as well :)

Comment (wkozyra95) [reply packages/eas-cli/src/commands/update/index.ts:514]:
```suggestion
        '--message and either --branch or --channel are required when updating in non-interactive mode unless --auto flag is specified ',
```
maybe sth like this?

Comment (byCedric) [reply packages/eas-cli/src/commands/update/index.ts]: Let's combine them:

```suggestion
      '--branch and --message, or --channel and --message are required when updating in non-interactive mode unless --auto is specified',
```

**Pull request merged**